src/main/resources/xlr_xldeploy/XLDeployClient.py

```diff
@@ -373,15 +373,21 @@ def update_ci_property(self, ci_id, ci_property, property_value):
             data[ci_property] = property_value
         self.update_ci(ci_id, json.dumps(data), 'json')
 
-    def add_ci_to_environment(self, env_id, ci_id):
+    def add_ci_to_environment(self, env_id, ci_id, ci_type):
         self.check_ci_exist(env_id, throw_on_fail=True)
         ci = self.get_ci(env_id, 'json')
         data = json.loads(ci)
-        data["members"].append(ci_id)
+        if str(ci_type) == 'udm.Dictionary':
+            data["dictionaries"].append(ci_id)
+        else:
+            data["members"].append(ci_id)
         self.update_ci(env_id, json.dumps(data), 'json')
 
         get_env_response = self.get_ci(env_id, 'xml')
```

Comment:
Kinda weird, but why is this call happening 2x? Once using `json`, once using `xml`. Shouldn't it just be enough to use the `json` code?

Reply:
I tested after removing the json and it does not work. pressed for time so did not go further. Can we add a issue and get to that later?

Reply:
No. Adding more technical debt to something that we can't explain isn't a good practice.
So what do you get when using the `json` only variant?